eslint-plugin/package.json

```diff
@@ -0,0 +1,26 @@
+{
+  "name": "eslint-plugin-zowe-explorer",
+  "version": "0.0.0",
+  "description": "Custom ESLint Rules for ZOWE Explorer",
```

Comment:
I am confused with the word plugin. For me, plugin is optional but somehow I feel like this is not optional.

Reply:
Eslint has several parts. Its core (the eslint package), plugins (which can provide linting rules) and a user configuration. 

If someone wants to create a new rule for ESLint, they need to create an eslint plugin with the rule. That is what the eslint-plugin folder is.

To tell eslint I want it to use the plugin, the plugin has to be specified in eslint config https://github.com/zowe/vscode-extension-for-zowe/pull/786/files#diff-e4403a877d80de653400d88d85e4801aR4 . And to tell eslint to use a rule from the plugin for linting, it also has to be specified in the eslint config https://github.com/zowe/vscode-extension-for-zowe/pull/786/files#diff-e4403a877d80de653400d88d85e4801aR7.

Reply:
Does that answer your questions @jellypuno ?

Reply:
If the team was interested in more details, I could do a small presentation and explain how today's javascript/typescript tooling works under the hood.